src/frontend/qt_sdl/AudioSettingsDialog.cpp

```diff
@@ -68,6 +76,35 @@ AudioSettingsDialog::AudioSettingsDialog(QWidget* parent) : QDialog(parent), ui(
     ui->txtMicWavPath->setEnabled(iswav);
     ui->btnMicWavBrowse->setEnabled(iswav);
 
+    bool isext = (Config::MicInputType == 1);
+    ui->cbMic->setEnabled(isext);
+
+    #if QT_VERSION >= QT_VERSION_CHECK(6,0,0)
+        const auto deviceInfos = QMediaDevices::audioInputs();
+        for (const QAudioDevice &deviceInfo : deviceInfos)
+        {
+            ui->cbMic->addItem(deviceInfo.description());
+        }
+        
+        if (Config::MicDevice == "")
+        {
+            Config::MicDevice = QMediaDevices::defaultAudioInput().description().toStdString();
+        }
+    #else
+        const auto deviceInfos = QAudioDeviceInfo::availableDevices(QAudio::AudioInput);
+        for (const QAudioDeviceInfo &deviceInfo : deviceInfos)
+        {
+            ui->cbMic->addItem(deviceInfo.deviceName());
+        }
+
+        if (Config::MicDevice == "")
+        {
+            Config::MicDevice = QAudioDeviceInfo::defaultInputDevice().deviceName().toStdString();
+        }
+    #endif
+    ui->cbMic->setCurrentText(QString(Config::MicDevice.c_str()));
```

Comment:
You can convert directly from `std::string` to `QString` without going through `c_str()`:
```suggestion
    ui->cbMic->setCurrentText(QString::fromStdString(Config::MicDevice));
```